src/sign.rs

```diff
@@ -1,61 +1,244 @@
 use crate::account;
-use anyhow::Result;
+use anyhow::{bail, Context, Result};
+use clap::{Args, Subcommand};
 use fuel_crypto::{Message, SecretKey, Signature};
 use fuel_types::Bytes32;
 use rpassword::prompt_password;
-use std::{path::Path, str::FromStr};
+use std::{
+    path::{Path, PathBuf},
+    str::FromStr,
+};
 
-fn sign_transaction(
-    tx_id: Bytes32,
-    account_index: usize,
-    password: &str,
-    path: &Path,
-) -> Result<Signature> {
-    let secret_key = account::derive_secret_key(path, account_index, password)?;
-    sign_transaction_with_private_key(tx_id, secret_key)
+/// Sign some data (e.g. a transaction ID, a file, a string, or a hex-string)
+/// using either a wallet account or a private key.
+#[derive(Debug, Args)]
+pub struct Sign {
+    /// Sign using the wallet account at the given index.
+    /// Uses a discrete interactive prompt for password input.
+    #[clap(long, value_name = "ACCOUNT_INDEX")]
+    pub account: Option<usize>,
+    /// Sign using a private key.
+    /// Uses a discrete interactive prompt for collecting the private key.
+    #[clap(long)]
+    pub private: bool,
+    /// Sign by passing the private key directly.
+    ///
+    /// WARNING: This is primarily provided for non-interactive testing. Using this flag is
+    /// prone to leaving your private key exposed in your shell command history!
+    #[clap(long)]
+    pub private_key: Option<SecretKey>,
+    /// Directly provide the wallet password when signing with an account.
+    ///
+    /// WARNING: This is primarily provided for non-interactive testing. Using this flag is
```

Comment:
Same as above.

Reply:
Same as above, for @nfurfaro's same as above 😆

Reply:
As above, the interactive password prompt is the default behaviour, this flag is only for non-interactive input.